### ISSUE TITLE
Chart update stops after 380 updates (new entries)

### ISSUE THREAD

**Issue by Diddlik** [beto-rodriguez/LiveCharts2#1464]
**Describe the bug**
I have a small example where I use line charts.
The example itself is very simple: a new value is periodically added to the graph and one is deleted at the beginning.

After 380 updates, the graph stops updating

**To Reproduce**

```
using CommunityToolkit.Mvvm.ComponentModel;
using LiveChartsCore;
using LiveChartsCore.SkiaSharpView;
using LiveChartsCore.SkiaSharpView.Painting;
using SkiaSharp;
using System.Collections.ObjectModel;

namespace chartSample;

public partial class MainWindowViewModel : ObservableObject
{
    //chart
    public ObservableCollection<ISeries> Series { get; set; }
    private ObservableCollection<int> observableValuesUpdateRate;
    [ObservableProperty] private Axis[] _xAxes;
    [ObservableProperty] private Axis[] _yAxes;
    [ObservableProperty] private int _counter = 0;
    const int _plotDataTotal = 250;
    public object Sync { get; } = new();

    private readonly Random _random = new();

    public MainWindowViewModel()
    {
        InitChart();
    }
    private async Task ReadData()
    {
        while (true)
        {
            await Task.Delay(100);
            lock (Sync)
            {
                observableValuesUpdateRate.Add(_random.Next(0, 100));
                if (observableValuesUpdateRate.Count > 250) observableValuesUpdateRate.RemoveAt(0);

                Counter++;
            }
        }
    }
    private void InitChart()
    {
        observableValuesUpdateRate = new ObservableCollection<int>();
        for (int i = 0; i < _plotDataTotal; ++i)
        {
            observableValuesUpdateRate.Add(0);
        }
        Series =
        [
            new LineSeries<int>
            {
                GeometryFill = null,
                Fill = new SolidColorPaint(SKColor.Parse("0097A9")),
                GeometryStroke = null,
                LineSmoothness = 0,
                Stroke = new SolidColorPaint(SKColor.Parse("0097A9"), 0.1f),
                Values = observableValuesUpdateRate,
            }

        ];

        var xAxis = new Axis
        {
            ShowSeparatorLines = false,
            IsVisible = false,
        };

        XAxes = [xAxis];

        var yAxis = new Axis
        {
            MaxLimit = 100,
            MinLimit = 0,
            ShowSeparatorLines = false,
            IsVisible = false

        };
        YAxes = [yAxis];
        _ = ReadData();
    }

}
```

and here the view:
```

<StackPanel>
 <Grid Grid.Row="1" Width="300" Background="DarkGray" Height="200" Margin="0,0,10,0" VerticalAlignment="Bottom" HorizontalAlignment="Right">
     <lvc:CartesianChart  
         SyncContext="{Binding Sync}"
         Series="{Binding Series}"
         TooltipPosition="Hidden"
         YAxes="{Binding YAxes}"
         XAxes="{Binding XAxes}">
     </lvc:CartesianChart>
 </Grid>
     <Label Content="{Binding Counter}" Foreground="AliceBlue" HorizontalAlignment="Right" Margin="0,10,10,0"/>
 </StackPanel>
```

**Screenshots**
![R7vo4D7jnh](https://github.com/beto-rodriguez/LiveCharts2/assets/6010713/748c8a26-8745-4cb8-9771-280628898bc9)

**Desktop (please complete the following information):**
 - OS: Windows, .NET 8, WPF

**Comment by Diddlik** [comment]
My findings till now: 
If I set on defining the axis (Y and X) IsVisible = true, update works fine, if one of axis has IsVisible = false it stops on about 380 updates.

**Comment by Diddlik** [comment]
Is it maybe possible to force chart update/ refresh?

**Comment by erdmenchen** [comment]
Im experiencing the same issue with 2.0.0-rc2 on .NET 8 on Android and iOS with chart not updating after a while.
I counted that it freezes after 420 updates on my mobile devices.

When it stops updating, it completely stops updating the UI, even changing visibilities of series does not work anymore or clearing the series completely.

> My findings till now: If I set on defining the axis (Y and X) IsVisible = true, update works fine, if one of axis has IsVisible = false it stops on about 380 updates.

I can confirm this. When showing all axis, it does not freeze.
 However this is not a solution for my use case 🙄

**Comment by erdmenchen** [comment]
This fixed it for me:
https://github.com/beto-rodriguez/LiveCharts2/issues/1495#issuecomment-2088313998 

Hopefully there is soon a new release that ships a fix.

**Comment by beto-rodriguez** [comment]
This is fixed with #1514, sorry for the delay on releasing that patch, a new version with this fix will be released in a few days, thanks for the issue, I will close this for now, we can keep track of this issue at #1495
